src/libraries/System.Text.Json/docs/KeyConverter_spec.md

```diff
@@ -0,0 +1,185 @@
+# Extend support for TKey in Dictionaries to non-string types
+
+# Motivation
+Most of our users that serialize dictionary use `Dictionary<string, TKey>`; however there is a significant amount that relies on `Dictionary<TKey, TValue>` where `TKey` is a primitive other than `string`, e.g: `int` or `Guid`, most of them came form `Newtonsoft.Json` which offers a plenty amount of support for using several types as the `TKey`, other popular .Net serializers also offer support for other types, the most common are integers (`int`, `uint`, `long`, etc.), and `enum`s (including Flags `enum`s).
+
+# Goals
+* 80%+ of dictionaries with non-string keys work out of the box, especially if they can round-trip.
+* Remain high performance.
```

Comment:
Add some more detail around using string property names?


- Use JSON property names for the key values. This produces more readable and smaller JSON and is consistent with `Newtonsoft.Json`.
  - However since property names must be a string, adds complexity since we now need a different mechanism to produce strings from non-string types.